bentoml/_internal/bento/build_config.py

```diff
@@ -362,6 +495,64 @@ def with_defaults(self) -> "CondaOptions":
 
 @attr.frozen
 class PythonOptions:
+    """
+    .. note::
+
+        Refers to :ref:`concepts/bento:Python Packages` section for more information.
+
+    Args:
+
+        requirements_txt (:code:`str`, `optional`):
+            Path to a custom :code:`requirements.txt` file.
+
+        packages: (:code:`list[str]`, `optional`):
+            List of pip packages to be installed in a given bento.
+
+            User can specify which packages to lock or not.
+
+            .. code-block:: yaml
+
+                python:
+                    packages:
+                    - "numpy"
+                    - "matplotlib==3.5.1"
+                    - "package>=0.2,<0.3"
+                    - "torchvision==0.9.2 --extra-index-url https://download.pytorch.org/whl/lts/1.8/cpu"
+                    - "git+https://github.com/bentoml/bentoml.git@main"
+
+        lock_packages: (:code:`bool`, `optional`):
+            Whether to lock the packages or not.
+
+            By default, BentoML will lock the packages for you using `pip-tools <https://github.com/jazzband/pip-tools>`_
```

Comment:
Would be useful to explain why locking packages is preferable and we do so by default, along the line of consistency between dev and prod and reproducibility.

Reply:
noted.